flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSink.java

```diff
@@ -170,114 +171,121 @@ public void testOverwriteTable() throws Exception {
+          SimpleDataUtil.createRecord(5, "d")
+      ));
+    } finally {
+      sql("DROP TABLE IF EXISTS %s.%s", flinkDatabase, tableName);
+    }
   }
 
   @Test
   public void testHashDistributeMode() throws Exception {
     String tableName = "test_hash_distribution_mode";
-
     Map<String, String> tableProps = ImmutableMap.of(
         "write.format.default", format.name(),
         TableProperties.WRITE_DISTRIBUTION_MODE, DistributionMode.HASH.modeName()
     );
     sql("CREATE TABLE %s(id INT, data VARCHAR) PARTITIONED BY (data) WITH %s",
         tableName, toWithClause(tableProps));
 
-    // Insert data set.
-    sql("INSERT INTO %s VALUES " +
-        "(1, 'aaa'), (1, 'bbb'), (1, 'ccc'), " +
-        "(2, 'aaa'), (2, 'bbb'), (2, 'ccc'), " +
-        "(3, 'aaa'), (3, 'bbb'), (3, 'ccc')", tableName);
-
-    Table table = validationCatalog.loadTable(TableIdentifier.of(icebergNamespace, tableName));
-    SimpleDataUtil.assertTableRecords(table, ImmutableList.of(
-        SimpleDataUtil.createRecord(1, "aaa"),
-        SimpleDataUtil.createRecord(1, "bbb"),
-        SimpleDataUtil.createRecord(1, "ccc"),
-        SimpleDataUtil.createRecord(2, "aaa"),
-        SimpleDataUtil.createRecord(2, "bbb"),
-        SimpleDataUtil.createRecord(2, "ccc"),
-        SimpleDataUtil.createRecord(3, "aaa"),
-        SimpleDataUtil.createRecord(3, "bbb"),
-        SimpleDataUtil.createRecord(3, "ccc")
-    ));
-
-    Assert.assertEquals("There should be only 1 data file in partition 'aaa'", 1,
-        SimpleDataUtil.partitionDataFiles(table, ImmutableMap.of("data", "aaa")).size());
-    Assert.assertEquals("There should be only 1 data file in partition 'bbb'", 1,
-        SimpleDataUtil.partitionDataFiles(table, ImmutableMap.of("data", "bbb")).size());
-    Assert.assertEquals("There should be only 1 data file in partition 'ccc'", 1,
-        SimpleDataUtil.partitionDataFiles(table, ImmutableMap.of("data", "ccc")).size());
-
-    sql("DROP TABLE IF EXISTS %s.%s", flinkDatabase, tableName);
+    try {
+      // Insert data set.
+      sql("INSERT INTO %s VALUES " +
+          "(1, 'aaa'), (1, 'bbb'), (1, 'ccc'), " +
+          "(2, 'aaa'), (2, 'bbb'), (2, 'ccc'), " +
+          "(3, 'aaa'), (3, 'bbb'), (3, 'ccc')", tableName);
+
+      Table table = validationCatalog.loadTable(TableIdentifier.of(icebergNamespace, tableName));
+      SimpleDataUtil.assertTableRecords(table, ImmutableList.of(
+          SimpleDataUtil.createRecord(1, "aaa"),
+          SimpleDataUtil.createRecord(1, "bbb"),
+          SimpleDataUtil.createRecord(1, "ccc"),
+          SimpleDataUtil.createRecord(2, "aaa"),
+          SimpleDataUtil.createRecord(2, "bbb"),
+          SimpleDataUtil.createRecord(2, "ccc"),
+          SimpleDataUtil.createRecord(3, "aaa"),
+          SimpleDataUtil.createRecord(3, "bbb"),
+          SimpleDataUtil.createRecord(3, "ccc")
+      ));
+
+      // Sometimes we will have more than one checkpoint if we pass the auto checkpoint interval
```

Comment:
maybe expand the comments a little more to provide more context to reader.
```
// As a result, we may produce multiple snapshots. So we need to assert that we only produces 1 file for every partition per snapshot
```

Reply:
Updated comment